Source/CarthageKit/Errors.swift

```diff
@@ -86,6 +86,15 @@ public enum CarthageError: Error, Equatable {
 
 	/// An error occurred while shelling out.
 	case taskError(TaskError)
+
+	/// An error in determining the local Swift version
+	case unknownLocalSwiftVersion
+
+	/// An error in determining the framework Swift version
+	case unknownFrameworkSwiftVersion
+
+	/// The framework binary is not compatible with the local Swift version.
+	case incompatibleFrameworkSwiftVersions(local: String, framework: String)
```

Comment:
The idea here was to have them all be user-facing. This regex-based parsing feels a little fragile to me. So since the check is pessimistic, this was intended to give the developer as much information as possible to diagnose why their framework isn't being used. Failing with these three error outputs:

```bash
*** Skipped installing BadOld.framework binary due to the error:
	"Incompatible Swift version - framework was built with 3.0 and the local version is 3.0.2."
```

```bash
*** Skipped installing BadOld.framework binary due to the error:
	"Unable to determine local Swift version."
```

```bash
*** Skipped installing BadOld.framework binary due to the error:
	"Unable to determine framework Swift version."
```

If you think this approach is undesirable, I can either reduce this verbosity down to a single error message, or I can create an internal error with these three failure modes and wrap them into a single `CarthageError`.